src/osip-monitoring-controller/src/main/java/edu/kit/pse/osip/monitoring/controller/MonitoringController.java

```diff
@@ -39,55 +56,350 @@
+            }
+            
+            for (TankContainer cont: tanks) {
+                TankClient old = cont.client;
+                hostname = currentSettings.getHostname(cont.tank.getTankSelector(), DEFAULT_HOSTNAME);
+                port = currentSettings.getPort(cont.tank.getTankSelector(), defaultPort++);
+                cont.client = new TankClient(new RemoteMachine(hostname, port));
+                cont.absClient = cont.client;
+                try {
+                    if (old != null) {
+                        old.disconnectClient();
+                    }
+                } catch (UAClientException e) {
+                    System.err.println("Error while disconnecting: " + e.getMessage());
+                }
+            }
+            mixCont.client.connectClient();
+            for (TankContainer cont : tanks) {
+                cont.client.connectClient();
+            }
+        } catch (UAClientException e) {
+            System.err.println("Unable to connect to servers. " + e.getMessage());
+            currentSettingsView.showCanNotConnectAlert();
+        }
     }
+    
+    private void handleSaveButton(ClientSettingsWrapper settings,
+            MonitoringViewInterface currentMonitoringViewInterface,
+            SettingsViewInterface currentSettingsViewInterface) {
+        int oldPort = -1;
+        boolean portChanged = true;
+        for (TankSelector tank: TankSelector.values()) {
+            oldPort = currentSettings.getPort(tank, -1);
+            if (currentSettingsViewInterface.getServerPort(tank) == oldPort) {
+                portChanged = false;
+            }
+            currentSettings.setFetchInterval(currentSettingsViewInterface.getUpdateInterval());
+            currentSettings.setFillLevelDiagram(tank, currentSettingsViewInterface.isFillLevelProgressionEnabled(tank));
+            currentSettings.setOverflowAlarm(tank, currentSettingsViewInterface.isOverflowAlarmEnabled(tank));
+            currentSettings.setOverheatingAlarm(
+                    tank, currentSettingsViewInterface.isTemperatureOverheatingAlarmEnabled(tank));
+            currentSettings.setServerHostname(tank, currentSettingsViewInterface.getServerHostname());
+            currentSettings.setServerPort(tank, currentSettingsViewInterface.getServerPort(tank));
+            currentSettings.setTempDiagram(tank, currentSettingsViewInterface.isTemperatureProgressionEnabled(tank));
+            currentSettings.setUndercoolingAlarm(
+                    tank, currentSettingsViewInterface.isTemperatureUndercoolingAlarmEnabled(tank));
+            currentSettings.setUnderflowAlarm(tank, currentSettingsViewInterface.isUnderflowAlarmEnabled(tank));
+            currentMonitoringViewInterface.setFillLevelProgressionEnabled(
+                    tank, currentSettingsViewInterface.isFillLevelProgressionEnabled(tank));
+            currentMonitoringViewInterface.setOverflowAlarmEnabled(
```

Comment:
Bitte aufteilen in 2 Methoden. Einmal nur `currentSettings.set` und einmal `currentMonitoringViewInterface.set`. Dann nur die mit `currentMonitoringViewInterface.set` im Konstruktor